src/script/integration/IntegrationRepository.ts

```diff
@@ -177,31 +201,33 @@ export class IntegrationRepository {
       : this.create1to1ConversationWithService(serviceEntity);
   }
 
-  getProviderById(providerId) {
+  getProviderById(providerId: string): Promise<any> {
     return this.integrationService.getProvider(providerId).then(providerData => {
       if (providerData) {
         return IntegrationMapper.mapProviderFromObject(providerData);
       }
+      return undefined;
     });
   }
 
-  getServiceById(providerId, serviceId) {
+  getServiceById(providerId: string, serviceId: string): Promise<any> {
     return this.integrationService.getService(providerId, serviceId).then(serviceData => {
       if (serviceData) {
         return IntegrationMapper.mapServiceFromObject(serviceData);
       }
+      return undefined;
     });
   }
 
-  getServices(tags, start) {
-    const tagsArray = _.isArray(tags) ? tags.slice(0, 3) : [ServiceTag.INTEGRATION];
+  getServices(tags: ServiceTag | ServiceTag[], start: string): Promise<ServiceEntity[]> {
+    const tagsArray = isArray(tags) ? tags.slice(0, 3) : [ServiceTag.INTEGRATION];
```

Comment:
Time to replace `underscore` with ES6 standards: https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray